pkg/git/libgit2/checkout.go

```diff
@@ -67,6 +67,22 @@ type CheckoutBranch struct {
 func (c *CheckoutBranch) Checkout(ctx context.Context, path, url string, opts *git.AuthOptions) (_ *git.Commit, err error) {
 	defer recoverPanic(&err)
 
+	if managed.Enabled() {
```

Comment:
My understanding is that we want to deprecate the git2go's default approach.
If that's the case, we should  ensure `transportAuthID` is never empty.